src/Infrastructure/Persistence/ParticipantDynamoDbRepository.cs

```diff
@@ -77,57 +82,105 @@ public async Task<ParticipantDetails> QueryIndexForParticipantDetailsAsync(strin
         {
             entity.Pk = ParticipantKey(nhsId);
             entity.Sk = ParticipantKey();
 
-            await _context.SaveAsync(entity, _config);
+            await _context.SaveAsync(entity, _config, cancellationToken);
         }
 
-        public async Task UpdateParticipantDemographicsAsync(ParticipantDemographics entity)
+        public async Task UpdateParticipantDemographicsAsync(ParticipantDemographics entity,
+            CancellationToken cancellationToken = default)
         {
-            await _context.SaveAsync(entity, _config);
+            await _context.SaveAsync(entity, _config, cancellationToken);
         }
 
-        public async Task DeleteParticipantDetailsAsync(ParticipantDetails entity)
+        public async Task DeleteParticipantDetailsAsync(ParticipantDetails entity,
+            CancellationToken cancellationToken = default)
         {
-            await _context.DeleteAsync(entity, _config);
+            await _context.DeleteAsync(entity, _config, cancellationToken);
         }
 
-        public async Task CreateAnonymisedDemographicParticipantDataAsync(ParticipantDetails entity)
+        public async Task CreateAnonymisedDemographicParticipantDataAsync(ParticipantDetails entity,
+            CancellationToken cancellationToken = default)
         {
-            await _context.SaveAsync(entity, _config);
+            await _context.SaveAsync(entity, _config, cancellationToken);
+        }
+
+        public async IAsyncEnumerable<Participant> GetAllAsync(
+            [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            var search = _context.ScanAsync<Participant>(null, _config);
+
+            while (!search.IsDone)
+            {
+                var page = await search.GetNextSetAsync(cancellationToken);
+                foreach (var item in page)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    yield return item;
+                }
+            }
+        }
+
+        public async Task<Participant> GetParticipantAsync(string pk, CancellationToken cancellationToken = default)
+        {
+            return await _context.LoadAsync<Participant>(pk, ParticipantKey(),
+                _config, cancellationToken);
+        }
+
+        public async Task DeleteParticipantAsync(Participant entity, CancellationToken cancellationToken = default)
+        {
+            await _context.DeleteAsync(entity, _config, cancellationToken);
+        }
+
+        public async Task CreateParticipantAsync(Participant entity, CancellationToken cancellationToken = default)
+        {
+            await _context.SaveAsync(entity, _config, cancellationToken);
+        }
+
+        public async IAsyncEnumerator<Participant> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+        {
+            await foreach (var participant in GetAllAsync(cancellationToken))
```

Comment:
You are iterating this enumerator and wrapping it in another async/await.

Move the implementation of `GetAllAsync()` in here.